include/mrdox/Metadata/TemplateParam.hpp

```diff
@@ -29,16 +47,152 @@ namespace mrdox {
 //
 /** A template parameter.
 */
+
+
+struct TemplateParamInfo;
+
+struct TemplateTypeParamInfo
+{
+    // default type for the type template parameter
+    std::optional<TypeInfo> Default;
+};
```

Comment:
Concept?

Reply:
If there's a SymbolID in there, you could lose the optional.